packages/app/src/device-file-system/commands/file-upload.command.ts

```diff
@@ -0,0 +1,117 @@
+import { DeviceFileSystemError } from "App/device-file-system/constants"
+
+export class FileUploadCommand extends BaseCommand {
+  constructor(
+    public deviceService: DeviceService,
+    public fileSystemService: FileSystemService
+  ) {
+    super(deviceService)
+  }
+
+  public async exec(
+    directory: string,
+    path: string
+  ): Promise<ResultObject<undefined>> {
+    let data: Buffer | Uint8Array
+
+    try {
+      data = await this.fileSystemService.readFile(path)
+    } catch (error) {
+      return Result.failed(
+        new AppError(
+          DeviceFileSystemError.FileUploadUnreadable,
+          `Uploading file: file ${path} can't be opened`
+        )
+      )
+    }
+
+    const fileSize = Buffer.byteLength(data)
+    const fileCrc32 = this.countCRC32(data)
+    const fileName = path.split("/").reverse()[0]
+
+    const response = await this.deviceService.request({
+      endpoint: Endpoint.FileSystem,
+      method: Method.Put,
+      body: {
+        fileSize,
+        fileCrc32,
+        fileName: [directory, fileName].join("/"),
+      },
+    })
+
+    if (
+      response.status !== RequestResponseStatus.Ok ||
+      response.data === undefined
+    ) {
+      return Result.failed(
+        new AppError(
+          DeviceFileSystemError.FileUploadRequest,
+          "Uploading file: Something went wrong in init sending request"
+        )
+      )
+    }
+
+    const { txID, chunkSize } = response.data
+    return this.sendFileRequest(data, txID, chunkSize)
+  }
+
+  private async sendFileRequest(
+    buffer: Buffer | Uint8Array,
+    txID: string,
+    chunkSize: number,
+    chunkNo = 1
+  ): Promise<ResultObject<undefined>> {
+    try {
+      const sliceStart = (chunkNo - 1) * chunkSize
+      const sliceEnd = sliceStart + chunkSize
+      const chunkedBuffer = buffer.slice(sliceStart, sliceEnd)
+      const chunkedBufferSize = Buffer.byteLength(chunkedBuffer)
+      const lastChunk = chunkedBufferSize < chunkSize
+
+      const response = await this.deviceService.request({
+        endpoint: Endpoint.FileSystem,
+        method: Method.Put,
+        body: {
+          txID,
+          chunkNo,
+          data: chunkedBuffer.toString("base64"),
+        },
+      })
+
+      if (response.status !== RequestResponseStatus.Ok) {
+        return Result.failed(
+          new AppError(
+            DeviceFileSystemError.FileUploadChunk,
+            "Uploading file: Something went wrong in sent chunk fie."
+          )
+        )
+      } else {
+        if (lastChunk) {
+          return Result.success(undefined)
+        }
+        return this.sendFileRequest(buffer, txID, chunkSize, chunkNo + 1)
+      }
+    } catch (error) {
+      return Result.failed(
+        new AppError(
+          DeviceFileSystemError.FileUploadChunk,
+          error
+            ? (error as Error).message
+            : "Uploading file: Something went wrong in sent chunk fie."
```

Comment:
```suggestion
            : "Uploading file: Something went wrong in sent chunk file."
```

Reply:
I'll fix it in next PR

Reply:
https://github.com/mudita/mudita-center/pull/906/commits/2c7dd2e1e48fd486028de7c64e2cf57e6fa08281